static/app/api.tsx

```diff
@@ -535,6 +535,7 @@ export class Client {
 
           const responseContentType = response.headers.get('content-type');
           const isResponseJSON = responseContentType?.includes('json');
+          const wasExpectingJson = headers.get('Content-Type') === 'application/json';
```

Comment:
Can we rename this variable so it's obvious how this is different than the `responseContentType`/`isResponseJSON` combo directly above it?

```suggestion
          const wasExpectingJson = requestHeaders.get('Content-Type') === 'application/json';
```